overlord/snapshotstate/snapshotstate.go

```diff
@@ -69,6 +77,70 @@ func allActiveSnapNames(st *state.State) ([]string, error) {
 	return names, nil
 }
 
+func automaticSnapshotExpiration(st *state.State) time.Duration {
+	// TODO: get from config
+	return defaultAutomaticSnapshotExpiration
+}
+
+// saveExpiration saves expiration date of the given snapshot set in the state.
+// The state needs to be locked by the caller.
+func saveExpiration(st *state.State, setID uint64, expiryTime time.Time) error {
+	var snapshots map[uint64]*snapshotState
+	err := st.Get("snapshots", &snapshots)
+	if err != nil && err != state.ErrNoState {
+		return err
+	}
+	if snapshots == nil {
+		snapshots = make(map[uint64]*snapshotState)
+	}
+	snapshots[setID] = &snapshotState{
+		Expiry: expiryTime,
+	}
+	st.Set("snapshots", snapshots)
+	return nil
+}
+
+// removeSnapshotState removes given set IDs from the state.
+func removeSnapshotState(st *state.State, setIDs ...uint64) error {
+	var snapshots map[uint64]*snapshotState
```

Comment:
this can use json.RawMessage given we don't care about the contents

Reply:
Yep, good point, will do.